test-junit5/src/main/java/io/micronaut/test/extensions/junit5/MicronautJunit5Extension.java

```diff
@@ -213,16 +221,42 @@ private TestContext buildContext(ExtensionContext context) {
 
     @Override
     public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
-        return applicationContext != null && applicationContext.containsBean(parameterContext.getParameter().getType());
+        final Argument<?> argument = getArgument(parameterContext, applicationContext);
+        if (argument != null) {
+            if (argument.isAnnotationPresent(Value.class) || argument.isAnnotationPresent(Property.class)) {
+                return true;
+            } else {
+                return applicationContext.containsBean(argument.getType(), resolveQualifier(argument));
```

Comment:
Note when Micronaut 3.0 is out there should be just `argument` and not `argument.getType()` to support generic types